third_party/xla_client/pjrt_computation_client.cc

```diff
@@ -106,9 +115,15 @@ std::vector<ComputationClient::ComputationPtr> PjRtComputationClient::Compile(
   std::vector<ComputationClient::ComputationPtr> computations;
 
   for (auto& instance : instances) {
+    PjRtDevice* pjrt_device = StringToPjRtDevice(instance.compilation_device);
     xla::ProgramShape program_shape =
         instance.computation.GetProgramShape().ValueOrDie();
     xla::CompileOptions compile_options;
+    // TODO(wcromar): set compile_options.argument_layouts, enable strict shapes
+    compile_options.executable_build_options.set_num_replicas(
+        client_->addressable_device_count());
+    compile_options.executable_build_options.set_device_ordinal(
+        pjrt_device->id());
     std::unique_ptr<xla::PjRtExecutable> executable =
```

Comment:
Would this be shared if we compile once and execute on multiple devices? I guess for the current `Execute` fn paradigm we don't have to, but would this assumption change later?